tests/test_phase3_integration.py

```
@@ -0,0 +1,1012 @@
+                self._circuit_breaker = CircuitBreaker(
+                    service_name="MockProvider:test://mock-provider",
+                    failure_threshold=kwargs.get("cb_failure_threshold", 3),
+                    recovery_timeout=kwargs.get("cb_recovery_timeout", 0.1),
+                    half_open_max_calls=kwargs.get("cb_half_open_max_calls", 2),
+                    error_classifier=self._is_error_retryable,
+                )
+
+            async def generate_content_async(self, prompt, model_name, **kwargs):
+                if self._should_fail:
+                    self._failure_count += 1
+                    raise Exception(f"Mock provider failure #{self._failure_count}")
```

Pylint (3.3.7)

[warning] Raising too general exception: Exception

(W0719)